src/commands/init.ts

```diff
@@ -0,0 +1,164 @@
+/* eslint-disable unicorn/no-await-expression-member */
+import {Errors, Flags} from '@oclif/core'
+import chalk from 'chalk'
+import {existsSync, renameSync, statSync} from 'node:fs'
+import {readdir, rm, writeFile} from 'node:fs/promises'
+import {join, resolve, sep} from 'node:path'
+
+import {FlaggablePrompt, GeneratorCommand, exec, makeFlags, readPJSON} from '../generator'
+import {validateBin} from '../util'
+
+async function clone(repo: string, location: string): Promise<void> {
+  try {
+    await exec(`git clone https://github.com/oclif/${repo}.git "${location}" --depth=1`)
+  } catch (error) {
+    const err =
+      error instanceof Error
+        ? new Errors.CLIError(error)
+        : new Errors.CLIError('An error occurred while cloning the template repo')
+    throw err
+  }
+}
+
+const validModuleTypes = ['ESM', 'CommonJS']
+const validPackageManagers = ['npm', 'yarn', 'pnpm']
+
+const FLAGGABLE_PROMPTS = {
+  bin: {
+    message: 'Command bin name the CLI will export',
+    validate: (d: string) => validateBin(d) || 'Invalid bin name',
+  },
+  'output-dir': {
+    message: 'Directory to build the CLI in',
+    validate: (d: string) => existsSync(resolve(d)),
+  },
+  'package-manager': {
+    message: 'Select a package manager',
+    options: validPackageManagers,
+    validate: (d: string) => validPackageManagers.includes(d) || 'Invalid package manager',
+  },
+} satisfies Record<string, FlaggablePrompt>
+
+export default class Generate extends GeneratorCommand<typeof Generate> {
+  static description =
+    'This will add the necessary oclif bin files, add oclif config to package.json, and install @oclif/core if needed.'
+
+  static examples = [
+    {
+      command: '<%= config.bin %> <%= command.id %>',
+      description: 'Initialize a new CLI in the current directory with auto-detected module type and package manager',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> --output-dir "/path/to/existing/project"',
+      description: 'Initialize a new CLI in a different directory',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> --module-type "ESM" --package-manager "npm"',
+      description: 'Supply answers for specific prompts',
+    },
+  ]
+
+  static flaggablePrompts = FLAGGABLE_PROMPTS
+
+  static flags = {
+    ...makeFlags(FLAGGABLE_PROMPTS),
+    'module-type': Flags.option({
+      options: validModuleTypes,
+    })({}),
+  }
+
+  static summary = 'Initialize a new oclif CLI'
+
+  async run(): Promise<void> {
+    const outputDir = await this.getFlagOrPrompt({defaultValue: './', name: 'output-dir', type: 'input'})
```

Comment:
I went back and forth on whether this should be a flag or an arg. I landed on flag to better match how `generate` works but I don't feel strongly either way.

Reply:
I think it's good to allow tools like this to operate in other directories, like `npm --prefix /path/to/project` or `git -C /path/to/project`